components/script/dom/bindings/conversions.rs

```diff
@@ -596,11 +596,16 @@ pub unsafe fn get_property_jsval(
         Ok(cname) => cname,
         Err(_) => return Ok(()),
     };
-    JS_GetProperty(cx, object, cname.as_ptr(), rval);
-    if JS_IsExceptionPending(cx) {
-        return Err(Error::JSFailed);
+    let mut found = false;
+    if JS_HasProperty(cx, object, cname.as_ptr(), &mut found) && found {
+        JS_GetProperty(cx, object, cname.as_ptr(), rval);
+        if JS_IsExceptionPending(cx) {
+            return Err(Error::JSFailed);
+        }
+        Ok(())
+    } else {
+        Err(Error::JSFailed)
```

Comment:
This now causes JSFailed to be returned when !found. We need to write more careful checks to avoid that.